### DISCUSSION TITLE
How to disable host header checks?

### DISCUSSION THREAD

**Discussion by thenewguy** [OpenPrinting/cups#955] [Q&A]
I've been unable to workaround host header checks running CUPS in a container on Docker Desktop on Windows with host networking.

I am using CUPS 2.4.7 on Ubuntu 24.04.

Works great locally from the machine. But when trying to share the printers you are stuck with errors like this:
```
Request from "localhost" using invalid Host: field "1.2.3.4:631".
Request from "localhost" using invalid Host: field "cups.local:631".
```
where `1.2.3.4` is the actual ip address.

I've tried setting server name, server alias, listen addresses. Any suggestions? Thanks!

**Comment by michaelrsweet** [comment]
Using "ServerAlias *" should be sufficient to allow any hostname, however over localhost there are restrictions because DNS rebinding attacks often try to redirect valid hostnames to 127.0.0.1 or ::1. You may need to bridge the networking for the Docker instance to avoid these sorts of problems.
